pymc_marketing/clv/models/beta_geo.py

```diff
@@ -351,3 +353,77 @@ def expected_num_purchases_new_customer(
         return (left_term * right_term).transpose(
             "chain", "draw", "t", missing_dims="ignore"
         )
+
+    def _distribution_new_customers(
+        self,
+        random_seed: Optional[RandomState] = None,
+        var_names: Sequence[str] = ("population_dropout", "population_purchase_rate"),
+    ) -> xr.Dataset:
+        with pm.Model():
+            a = pm.HalfFlat("a")
+            b = pm.HalfFlat("b")
+            alpha = pm.HalfFlat("alpha")
+            r = pm.HalfFlat("r")
+
+            # This is the shape with fit_method="map"
+            if self.idata.posterior.dims == {"chain": 1, "draw": 1}:
```

Comment:
We should use the property because it will raise the informative error in case the model was not fit yet

```suggestion
            if self.fit_result.dims == {"chain": 1, "draw": 1}:
```